packages/editor/src/browser/editor-decoration.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2017 TypeFox and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+import { injectable, inject } from "inversify";
+import { Range } from "./editor";
+import { EditorManager } from "./editor-manager";
+import URI from "@theia/core/lib/common/uri";
+
+export const EditorDecorationsService = Symbol('EditorDecorationsService');
+
+export interface EditorDecorationsService {
+    registerDecorationType(decorationType: DecorationType): void;
+    removeDecorationType(key: string): void;
+    getDecorationTypes(): DecorationType[];
+    setDecorations(uri: URI, key: string, decorations: DecorationOptions[]): void;
+}
+
+@injectable()
+export class EditorDecorationsServiceImpl implements EditorDecorationsService {
```

Comment:
please check that files named after the main exported type, i.e. `editor-decoration-service.ts`